lib/commons/dom/is-visible.js

```diff
@@ -13,14 +13,187 @@ const clipRegex = /rect\s*\(([0-9]+)px,?\s*([0-9]+)px,?\s*([0-9]+)px,?\s*([0-9]+
 const clipPathRegex = /(\w+)\((\d+)/;
 
 /**
- * Determines if an element is hidden with a clip or clip-path technique
- * @method isClipped
+ * Determine whether an element is visible
+ * @method isVisible
  * @memberof axe.commons.dom
- * @private
- * @param  {CSSStyleDeclaration} style Computed style
- * @return {Boolean}
+ * @instance
+ * @param {HTMLElement} el The HTMLElement
+ * @param {Boolean} screenReader When provided, will evaluate visibility from the perspective of a screen reader
+ * @param {Boolean} recursed
+ * @return {Boolean} The element's visibilty status
+ */
+export default function isVisible(el, screenReader, recursed) {
+  if (!el) {
+    throw new TypeError(
+      'Cannot determine if element is visible for non-DOM nodes'
+    );
+  }
+
+  const vNode = el instanceof AbstractVirtualNode ? el : getNodeFromTree(el);
+  el = vNode ? vNode.actualNode : el;
+
+  const cacheName = '_isVisible' + (screenReader ? 'ScreenReader' : '');
+  const { DOCUMENT_NODE, DOCUMENT_FRAGMENT_NODE } = window.Node ?? {};
+  const nodeType = vNode ? vNode.props.nodeType : el.nodeType;
+  const nodeName = vNode ? vNode.props.nodeName : el.nodeName.toLowerCase();
+
+  if (vNode && typeof vNode[cacheName] !== 'undefined') {
+    return vNode[cacheName];
+  }
+
+  if (nodeType === DOCUMENT_NODE) {
+    return true;
+  }
+
+  const hiddenMethods = [elementHidden];
+  let style;
+  if (screenReader) {
+    hiddenMethods.push(ariaHidden);
+  }
+
+  if (el) {
+    if (nodeType === DOCUMENT_FRAGMENT_NODE) {
+      el = el.host; // grab the host Node
+    }
+
+    style = window.getComputedStyle(el, null);
+    if (style === null) {
+      return false;
```

Comment:
How is this possible?

Reply:
Seems to be an old Firefox bug that [has been in the code for a decade](https://github.com/dequelabs/axe-core/commit/b1d4b027f46a7f9d8d8cf7b600fea91f8e40830d#diff-3e7200e9a058efa36f66332b3f68b47ff659a5bf0241a1a1dd871eda19ae8ce0R39). I checked latest firefox and IE11 and they both seem to return a style object and not `null`.